code/__HELPERS/__thread_monitoring.dm

```diff
@@ -0,0 +1,120 @@
+//We wrap all procs that sleep so we can monitor how many threads are active
+//sadly we can't catch the entry point of input verbs but we should be working to clear those oout where possible
+
+#define SLEEP(X)\
+    ++Master.sleeping_threads;\
+    sleep(X);\
+    --Master.sleeping_threads;\
+    world.SleepEnd(world.time - X);
+
+/proc/sleep_proc(X) //for places the define can't be used. *coughcough*world.dm

```

Comment:
Why not just move world.dm and _compile_options.dm into a new subfolder of code?